rest/data/project_settings.go

```diff
@@ -241,6 +241,10 @@ func SaveProjectSettingsForSection(ctx context.Context, projectId string, change
 			}
 		}
 
+		if !mergedSection.IsRepotrackerDisabled() && mergedSection.RemotePath == "" {
```

Comment:
Actually since project repotrackers are enabled by default, this should probably also check IsEnabled

Reply:
I think it's okay if their settings get in a bad state, since they haven't enabled the project yet it won't actually matter until they enable it (at which case we will then error). The project technically starts in a bad state (since it starts with repotracker enabled and config undefined).